src/clickonce/launcher/HostFinder.cs

```diff
@@ -69,18 +110,26 @@ private static string GetProgramFilesFolder(bool is64bit = false)
         /// </summary>
         /// <param name="is64bit">If 64-bit bitness is required</param>
         /// <returns></returns>
-        private static string GetGlobalHost(bool is64bit = false)
+        private string GetGlobalHost(bool is64bit = false)
```

Comment:
Yeah, that's one option I considered, but went with instance and private field for architecture, as a way to avoid changing the existing methods. I'm also calling GetX86Host and Get64bitHost in sequence, for MSIL apps, as I'm interested in 64-bit native location, and not interested in which architecture it is.

I'll play with that option and simplify code to pass the architecture.

Reply:
Fixed this in new commit.